### ISSUE TITLE
Can't configure hugepages kernel args, image factory changes order in kernel command line.

### ISSUE THREAD

**Issue by kdish** [siderolabs/image-factory#154]
Trying to generate a bare metal image of talos 1.8.1 with the following extra kernel args
`hugepagesz=1G hugepages=32 hugepagesz=2M hugepages=1024`

All arguments entered are present in the generated kernel command line (and image), but they are not in the same order. (The hugepage settings require that the size and count are in sequence).

The resulting kernel command line  from image factory looks like this:
`talos.platform=metal console=tty0 init_on_alloc=1 slab_nomerge pti=on consoleblank=0 nvme_core.io_timeout=4294967295 printk.devkmsg=on ima_template=ima-ng ima_appraise=fix ima_hash=sha512 hugepagesz=1G hugepagesz=2M hugepages=32 hugepages=1024`

This results in hugepages being disabled at boot.

**Comment by smira** [comment]
Yes, this is a bug, but partially also the way Talos handles kernel command line, so fixing it won't be easy, but we should fix that.

**Comment by kdish** [comment]
Thanks for accepting the report! Is there any possible workaround i could try? I'm trying to run an application that needs these hugepage settings and I would prefer to do it with talos....

**Comment by kdish** [comment]
FYI, I managed to get hugepages working by mounting the /boot partition from the generated and editing the kernel command line in grub/grub.cfg
After booting on my modified image I get:
`$ kubectl describe nodes | grep -A 8 Allocatable | grep huge
  hugepages-1Gi:      32Gi
  hugepages-2Mi:      2Gi`

Which is what I want...
Please let me know if there are any fundamental problems with this approach, otherwise I will go with this until the bug is fixed.

**Comment by smira** [comment]
Yes, this is the only workaround unless there's any other way to configure hugepages.

**Comment by smira** [comment]
https://docs.kernel.org/admin-guide/mm/hugetlbpage.html there seems to be multiple ways to configure hugepages, so it might be that you can get around by using sysfs and machine configuration.

**Comment by kdish** [comment]
Thanks. I'll look into the docs.

**Comment by jleeh** [comment]
If anyone is having this issue, a workaround is as follows in the machineconfig:

```
  install:
    extraKernelArgs:
      - hugepagesz=1G
      - hugepages=4
  sysctls:
    vm.nr_hugepages: "2048"
```

`nr_hugepages` sysctl will set the default hugepages amount which is 2Mi, and then the kernel args can only be specified once to set extra hugepage sizes.

Due to the lib used within Talos to merge the cmd arguments, if there is duplicate arguments it will change the ordering which is what is causing this behaviour.
